src/core/agent/services/credentialService.ts

```diff
@@ -179,6 +169,15 @@ class CredentialService extends AgentService {
     })
   }
 
+  async removeCredentialsPendingDeletion(id: string) {
```

Comment:
This isn't right - please check how it's done for identifiers. `removeIdentifiersPendingDeletion` will fetch all identifiers pending deletion and then delete them.

This is updating a specific credential to not be pending deletion so it doesn't make sense.

Reply:
You also need to call the function when we come online in `markAgentStatus`